pkg/apis/logging/v1alpha1/temp.go

```diff
@@ -0,0 +1,40 @@
+package v1alpha1
+
+//this file is temporary and should be removed once we correct
+//the remaining code
+
+import (
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (cluster *ClusterLogging) Type() metav1.TypeMeta {
+	return cluster.TypeMeta
+}
+func (cluster *ClusterLogging) Meta() metav1.ObjectMeta {
+	return cluster.ObjectMeta
+}
+
+func (cluster *ClusterLogging) SchemeGroupVersion() string {
+	return SchemeGroupVersion.String()
+}
+
+func DoesClusterLoggingExist(cluster *ClusterLogging) bool {
+
+	// this is to avoid an invalid memory address panic
+	if cluster == nil {
+		return false
+	}
+
+	if err := sdk.Get(cluster); err != nil {
```

Comment:
didn't we want to avoid updating the ClusterLogging object? this will do that since we pass it as a pointer.
I think we want to do a deepcopy of it and then `get` on that object

Reply:
We do that here: https://github.com/openshift/cluster-logging-operator/pull/85/files#diff-156742c24dc12665a82a7fed5cf7c8abR57   I'll update in this method too